classes/issue/Collector.inc.php

```diff
@@ -0,0 +1,408 @@
+<?php
+
+namespace APP\issue;
+
+use HookRegistry;
+use Illuminate\Database\Query\Builder;
+use Illuminate\Support\Facades\DB;
+use PKP\core\interfaces\CollectorInterface;
+use PKP\core\PKPApplication;
+
+class Collector implements CollectorInterface
+{
+    public const ORDERBY_DATE_PUBLISHED = 'datePublished';
+    public const ORDERBY_LAST_MODIFIED = 'lastModified';
+    public const ORDERBY_SEQUENCE = 'seq';
+    public const ORDERBY_PUBLISHED_ISSUES = 'publishedIssues';
+    public const ORDERBY_UNPUBLISHED_ISSUES = 'unpublishedIssues';
+    public const ORDERBY_SHELF = 'shelf';
+    public const ORDER_DIR_ASC = 'ASC';
+    public const ORDER_DIR_DESC = 'DESC';
+    private const ORDER_CURRENT_ISSUE = 'currentIssue';
+
+    public DAO $dao;
+
+    public ?int $count = null;
+
+    public ?int $offset = null;
+
+    /** @var array|null Context ID or PKPApplication::CONTEXT_ID_ALL to get from all contexts */
+    public ?array $contextIds = null;
+
+    /** @var array|null order and direction pairing for queries */
+    public ?array $resultOrderings = null;
+
+    /** @var boolean|null return published issues */
+    public ?bool $isPublished = null;
+
+    /** @var array|null return issues in volume(s) */
+    public ?array $volumes = null;
+
+    /** @var array|null return issues with number(s) */
+    public ?array $numbers = null;
+
+    /** @var array|null return issues with year(s) */
+    public ?array $years = null;
+
+    /** @var array|null return issues that match a title */
+    public ?array $titles = null;
+
+    /** @var string|null Returns Issue by URL path  */
+    public ?string $urlPath = null;
+
+    /** @var string|null return issues which match words from this search phrase */
+    public ?string $searchPhrase = null;
+
+    public function __construct(DAO $dao)
+    {
+        $this->dao = $dao;
+    }
+
+    /**
+     * Set context issues filter
+     *
+     * @return $this
+     */
+    public function filterByContextIds(?array $contextIds): self
+    {
+        $this->contextIds = $contextIds;
+        return $this;
+    }
+
+    /**
+     * Set result order column
+     *
+     * @return $this
+     */
+    public function orderBy(string $column): self
```

Comment:
`$column` no longer refers to a column name -- it's a set of defined constants -- so I'd suggest changing its name. Also the self-documentation.